src/Microsoft.Identity.Client/TelemetryTokenCacheAccessor.cs

```diff
@@ -0,0 +1,92 @@
+﻿//----------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using Microsoft.Identity.Client.Internal;
+
+namespace Microsoft.Identity.Client
+{
+    internal class TelemetryTokenCacheAccessor : TokenCacheAccessor
+    {
+        // The content of this class has to be placed outside of its base class TokenCacheAccessor,
+        // otherwise we would have to modify multiple implementations of TokenCacheAccessor on different platforms.
+        public void SaveAccessToken(string cacheKey, string item, RequestContext requestContext)
+        {
+            var cacheEvent = new CacheEvent(CacheEvent.TokenCacheWrite) { TokenType = CacheEvent.TokenTypes.AT };
+            Telemetry.GetInstance().StartEvent(requestContext.TelemetryRequestId, cacheEvent);
```

Comment:
Guess what? That was exactly one of my earlier attempt when working on a local proof-of-concept.

    $ git lg1|grep using
    | | * 4a52e3a - (4 weeks ago) Experimental internal API based on "using" - Ray Luo (telemetry-using)

Later I/we shifted gear into heads-down-coding mode, and anything fancy and/or not-applicable-crossing-platforms would potentially cause lots of discussion, so a more old-school way was more proper at that time.

We would probably revisit some of the design/implementation decisions after the Build.